### PR TITLE
Move Blas state into THCState.

### PR DESCRIPTION
**Pull request by dominikgrewe**
Moves the last remaining global state (as far as I'm aware) into THCState.

### CODE REVIEW

File: lib/THC/THCGeneral.c [hunk 1]
```diff
@@ -13,6 +14,7 @@ void THCudaInit(THCState* state)
   state->rngState = (THCRNGState*)malloc(sizeof(THCRNGState));
   THCRandom_init(state, count, device);
 
+  state->blasState = (THCBlasState*)malloc(sizeof(THCBlasState));
```
Comment (soumith) [inline lib/THC/THCGeneral.c]: you need to free the state on shutdown in line 42

Comment (dominikgrewe) [inline lib/THC/THCGeneral.c]: Done!

File: lib/THC/THCGeneral.h [hunk 2]
```diff
@@ -25,11 +25,13 @@
 #endif
 
 struct THCRNGState;  /* Random number generator state. */
+struct THCBlasState;
 
 /* Global state to be held in the cutorch table. */
 typedef struct THCState
 {
   struct THCRNGState* rngState;
+  struct THCBlasState* blasState;
```
Comment (soumith) [inline lib/THC/THCGeneral.h]: just also noticed. this change has to be also reflected in FFI.lua

Comment (soumith) [inline lib/THC/THCGeneral.h]: FFI.lua line 8

Comment (dominikgrewe) [inline lib/THC/THCGeneral.h]: Good catch, thanks! I keep forgetting about FFI :s

**Comment by soumith** [comment]
great, looks good. squash your commits and will merge. thanks.

**Comment by dominikgrewe** [comment]
I squashed the commits.

**Pull request merged**